includes/Modules/AdSense.php

```diff
@@ -838,4 +858,84 @@ public function check_service_entity_access() {
 		return true;
 	}
 
+	/**
+	 * Validates the report metrics.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $metrics The metrics to validate.
+	 *
+	 * @return null|WP_Error null if metrics are valid, otherwise WP_Error.
+	 */
+	protected function validate_report_metrics( $metrics ) {
+		if ( false === $this->is_using_shared_credentials ) {
+			return null;
+		}
+
+		$valid_metrics = apply_filters(
+			'googlesitekit_shareable_adsense_metrics',
+			array(
+				'ESTIMATED_EARNINGS',
+				'IMPRESSIONS',
+				'PAGE_VIEWS_CTR',
+				'PAGE_VIEWS_RPM',
+			)
+		);
+
+		$invalid_metrics = array_diff( $metrics, $valid_metrics );
+
+		if ( count( $invalid_metrics ) > 0 ) {
+			return sprintf(
+				/* translators: %s is replaced with a comma separated list of the invalid metrics. */
+				_n(
+					'Unsupported metric requested: %s',
+					'Unsupported metrics requested: %s',
+					count( $invalid_metrics ),
+					'google-site-kit'
+				),
+				implode( ', ', $invalid_metrics )
+			);
+		}
+
+		return null;
+	}
+
+	/**
+	 * Validates the report dimensions.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $dimensions The dimensions to validate.
+	 *
+	 * @return null|WP_Error null if dimensions are valid, otherwise WP_Error.
+	 */
+	protected function validate_report_dimensions( $dimensions ) {
+		if ( false === $this->is_using_shared_credentials ) {
+			return null;
+		}
+
+		$valid_dimensions = apply_filters(
+			'googlesitekit_shareable_adsense_dimensions',
+			array(
+				'DATE',
```

Comment:
As per https://github.com/google/site-kit-wp/issues/5712#issuecomment-1222341102, please take the time to double-check the codebase and verify this dimension is the only one in current use.